pkg/codegen/schema/schema_test.go

```diff
@@ -2039,3 +2039,29 @@ func TestRoundtripAliasesYAML(t *testing.T) {
 
 	assert.YAMLEq(t, string(schemaBytes), string(yamlData))
 }
+
+func TestDanglingReferences(t *testing.T) {
+	t.Parallel()
+
+	testdataPath := filepath.Join("..", "testing", "test", "testdata")
+	loader := NewPluginLoader(utils.NewHost(testdataPath))
+	pkgSpec := readSchemaFile("dangling-reference-bad.json")
+	_, diags, _ := BindSpec(pkgSpec, loader, SchemaValidationOptions{})
```

Comment:
Maybe add a test that `allowDanglingReferences: true` does not result in an error? Also maybe for this (and the below) test setting `allowDanglingReferences: false` in the struct might be valuable to show what we're testing (it's the default of course, but I think it communicates the intent)